modules/openxr/extensions/openxr_opengl_extension.cpp

```diff
@@ -28,7 +28,7 @@
 /* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
 /*************************************************************************/
 
-#ifdef GLES3_ENABLED
+#if defined(GLES3_ENABLED) && defined(X11_ENABLED)
```

Comment:
I would be very surprised if openxr required X11 support. Are the changes to the OpenXR files needed to fix the reported bug?

Reply:
On Linux, presently, OpenGL support in OpenXR does depend on X11. (The plan is to fix that after we have Wayland support.) But this code change isn't quite right because I think it'd prevent OpenXR building on Windows and Android too. Something more targetted is needed.

Reply:
@clayjohn Yes, without this change the build with no X11 fails :(

@dsnopek Thanks for pointing this out, I will check for a better solution. Any suggestions are welcome of course ;)

Reply:
@pfertyk Maybe add some logic in the SCsub to skip the whole file?

https://github.com/godotengine/godot/blob/master/modules/openxr/SCsub#L93

You'd still need a preprocessor directive to prevent the extension getting registered, though:

https://github.com/godotengine/godot/blob/master/modules/openxr/openxr_api.cpp#L1254